### ISSUE TITLE
angular preset recommends patch bump when no commits

### ISSUE THREAD

**Issue by adanilev** [conventional-changelog/conventional-changelog#399]
The conventional-changelog-angular package recommends a patch bump even when there are no commits. Is this intentional? If not, happy to raise a PR to resolve.

**Comment by jamesovens** [comment]
I am seeing this also. Other tools such as semantic-release seem to accurately report when there are no changes. This is slightly annoying for us as we would like to determine when there are no changes that require a release and in that scenario we will choose to not go through the release and deployment process.

**Comment by tommywo** [comment]
I think this is because of [usage in standard-version](https://github.com/conventional-changelog/standard-version/blob/106d0390f7fe73c93131511caef17f2dca37fd6a/lib/lifecycles/bump.js#L143), which is meant to be used when a person wants to bump a version. That could be changed in **presets** so whatBump() would return -1, in **conventional-recommended-bump** returning `recommendation.releaseType = 'none'`  and in **standard-version** changing this to `patch` unless additional flag f.e. `skipBumpOnlyRelease` is `true` and in that case exiting without an error.
Note that that would be a breaking change for **presets** and **conventional-recommended-bump**.

**Comment by jamesovens** [comment]
Is this supposed to only be a problem in the angular preset? Because I tried the conventionalcommits preset instead and it still happens with that:
https://github.com/conventional-changelog/conventional-changelog/tree/master/packages/conventional-changelog-conventionalcommits

**Comment by rrundle-dave** [comment]
Yeah this is because the default level in the[`whatBump` function is 2](https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-angular/conventional-recommended-bump.js#L9), which is a patch. So with an empty commit array it just returns that level.

**Comment by dangreen** [comment]
Thanks for the report. This is the longstanding behavior of the Angular preset: it defaults to a patch recommendation, including when the selected range is empty. We keep that behavior for compatibility.

If your release workflow needs to skip releases without bump-worthy changes, make that decision in the calling tool based on the recommendation, or use the conventionalcommits preset, which returns no release type when no bump-effect commits are present. We do not plan to change the Angular preset default, so closing as not planned.
